firebase-perf/src/main/java/com/google/firebase/perf/FirebasePerfEarly.java

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+//
+// You may obtain a copy of the License at
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.perf;
+
+import android.content.Context;
+import android.os.Handler;
+import android.os.Looper;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.StartupTime;
+import com.google.firebase.perf.application.AppStateMonitor;
+import com.google.firebase.perf.config.ConfigResolver;
+import com.google.firebase.perf.metrics.AppStartTrace;
+import com.google.firebase.perf.session.SessionManager;
+
+/**
+ * The Firebase Performance early initialization.
+ *
+ * <p>Responsible for initializing the AppStartTrace, and early initialization of ConfigResolver
+ */
+public class FirebasePerfEarly {
+  private final Handler mainHandler = new Handler(Looper.getMainLooper());
+
+  public FirebasePerfEarly(FirebaseApp app) {
+    Context context = app.getApplicationContext();
+
+    // Initialize ConfigResolver early for accessing device caching layer.
```

Comment:
@vkryachko I wonder if we could merge the 2-part initialization (`FirebasePerfEarly` + `FirebasePerfRegistrar`) to simplify it further in the future. The reason we needed `FirebasePerfProvider` was to capture app-start time, but now that's no longer needed, and if Fireperf is completely started by component framework, if we get rid of `FirebasePerfEarly` then we also don't need `ConfigResolver` anymore. That's because `ConfigResolver` and its related classes were created when the old `FirebasePerfProvider` early initializes part of Fireperf that needed FRC, so we had to build our own disk cache for RC. If Fireperf starts after FRC as it should, we can just use FRC. However I'm not sure if this will be okay for the SDK-Observability-project, would it be fine for Fireperf's dependencies to start first?

Reply:
I don't have full context on why the 2-part logic was in place, but my understanding that part of it was to speed up the SDK and delay a lot of work as much as possible(i.e. see #2518).

But I trust your judgement to evolve this in a way that makes sense, after this initial migration. Happy to discuss it more.